openfl/interface/interactive_api/experiment.py

```diff
@@ -252,6 +264,21 @@ def _prepare_plan(self, model_provider, task_keeper, data_loader,
         # We also could change the aggregator logic so it will send tasks to aggregator
         # as soon as it connects. This change should be a part of a bigger PR
         # brining in fault tolerance changes
+
+        # Check tasks type
+        for name in task_keeper.task_registry:
+            if task_keeper.task_contract[name]['optimizer'] is not None:
+                self.train_task_exist = True
+            else:
+                self.validation_task_exist = True
```

Comment:
This logic divides all tasks into only two types (train and validate), and the only defining parameter is the presence of an optimizer. And in a very implicit way for user. Looks like we have to rethink it and separate it explicitly, or hang it with warnings.